studio/app/common/routers/workflow.py

```diff
@@ -87,3 +88,28 @@ async def import_workflow_config(file: UploadFile = File(...)):
         return WorkflowConfig(**contents)
     except Exception as e:
         raise HTTPException(status_code=400, detail=f"Parsing yaml failed: {str(e)}")
+
+
+@router.get(
+    "/sample_data",
+    dependencies=[Depends(is_workspace_available)],
+)
+async def copy_sample_data(workspace_id: str):
+    all_files_copied = True  # Initialize flag
+    folders = ["input", "output"]
+    for folder in folders:
+        sample_data_dir = os.path.join("sample_data", folder)
+        if folder == "input":
+            user_dir = os.path.join(DIRPATH.INPUT_DIR, workspace_id)
+        elif folder == "output":
+            user_dir = os.path.join(DIRPATH.OUTPUT_DIR, workspace_id)
+        if not os.path.exists(user_dir):
+            os.makedirs(user_dir)
+        try:
+            shutil.copytree(sample_data_dir, user_dir, dirs_exist_ok=True)
+        except Exception as e:
+            raise HTTPException(
+                status_code=500, detail=f"Failed to copy files from {folder}: {e}"
+            )
+            all_files_copied = False
```

Comment:
this code can be unreached.
(on raise HTTPException, the code would be finished.)

You don't have to use all_files_copied flag.
If any exception, just raise HTTPException, if not return True.